echopype/convert/set_groups_base.py

```diff
@@ -245,7 +244,9 @@ def _add_beam_dim(
                     .expand_dims(dim={"beam": np.array(["1"], dtype=str)}, axis=ds[var_name].ndim)
                     .copy()
                 )
-                ds[var_name].beam.attrs = self._varattrs["beam_coord_default"]["beam"]
+                ds[var_name].beam.attrs = sonarnetcdf_1.yaml_dict["variable_and_varattributes"][
+                    "beam_coord_default"
+                ]["beam"]
```

Comment:
With this change it seems line 47 `self._varattrs = sonarnetcdf_1.yaml_dict["variable_and_varattributes"]` should be removed. It is not used anywhere in the code.

Reply:
Line 47 should not be removed. `self._varattrs` is used in the specific `set_groups_X.py` for the sensors. It is necessary to use this line because this method should be static. We can't use `self._varattrs` here because this function is used in my conversion routines, which do not initialize this group (thus `self._varattrs` does not exist).

Reply:
Ah I see, thanks for the explanation!